src/opts.rs

```diff
@@ -150,4 +160,6 @@ pub struct Opts {
     pub concurrent_resolve: usize,
     #[structopt(flatten)]
     pub transfer_config: TransferConfig,
+    #[structopt(flatten)]
+    pub rewrite_config: RewriteConfig,
```

Comment:
Is it necessary to set rewrite config in cli? From my perspective, only some repos need hard-coded rewrite configurations.

Reply:
Users may want to use their own rewriting rules to adapt their own mirror service.

For example, SJTUG may safely specify something like `--rewrite-pattern https://download.haskell.org/ --rewrite-target https://mirrors.sjtu.edu.cn/haskell/`, but users may want to specify their own domains or rewriting rules.

Reply:
From my perspective, this kind of config should be part of the subcommand. e.g. `./mirror-clone ghc-up --target= https://mirrors.sjtu.edu.cn/haskell/`. And we developers should decide on what pattern to rewrite.